src/lib/utils/arweave.ts

```diff
@@ -0,0 +1,50 @@
+import Arweave from 'arweave';
+import { decrypt } from 'ecies-geth';
+import { utils } from 'ethers';
+
+export const initArweave = () => {
+  return Arweave.init({
+    host: 'arweave.net', // Hostname or IP address for a Arweave host
+    port: 443, // Port
+    protocol: 'https', // 'https', // Network protocol http or https
+    timeout: 20000, // Network request timeouts in milliseconds
+    logging: false, // Enable network request logging
+  });
+};
+
+export const fetchArweaveFileFallback = async (arweaveTxId: string): Promise<any> => {
+  try {
+    const arweave = initArweave();
+    const res = await arweave.api.get(`/${arweaveTxId}`, {
+      responseType: 'arraybuffer',
+    });
+    return res.data;
+  } catch (error) {
+    throw new Error(`Error fetching arweave file: ${error}`);
+  }
+};
+
+export const fetchArweaveFile = async (arweaveTxId: string): Promise<Uint8Array> => {
+  try {
+    const arweave = initArweave();
+    const data = await arweave.transactions.getData(arweaveTxId);
+
+    return data as Uint8Array;
+  } catch (error) {
+    throw new Error(`Error fetching arweave file: ${error}`);
+  }
+};
+
+export const decryptArweaveFile = async (data: Uint8Array, privateKey: string): Promise<Buffer> => {
```

Comment:
This may not be used. I'm using a decrypt function located in the helper file where the encrypt function is located.

Reply:
Looks like its being used by the `isArweaveFileValid` function --- do we need to validate the arweave file at any point? If so, could add a TODO that the validate function needs to be used somewhere. Otherwise could nuke both this function and the validate one.

Reply:
I don't think we have a need to validate the arweave file... Once it's fetched and decrypted the work is already done.